src/Castle.Windsor.Tests/Facilities/TypedFactory/TypedFactoryAndSubContainersTestCase.cs

```diff
@@ -51,12 +51,31 @@ public void Facility_When_added_to_a_child_container_wher_parent_has_facility_pu
 		}
 
 		[Test]
+		public void FactoryCreatesAnotherSingletonFactory_FactoryReleased_ShouldNotDisposeSingleton()
+		{
+			Container.Kernel.AddFacility<TypedFactoryFacility>();
+			Container.Register(
+				Component.For<A>().LifestyleTransient(),
+				Component.For<IGenericFactory<IGenericFactory<A>>>().AsFactory().LifestyleTransient(),
+				Component.For<IGenericFactory<A>>().AsFactory().LifestyleSingleton());
```

Comment:
The comments in #61 indicate that you don't need nested factories to reproduce this defect, we should include another test with a singleton service too.

Reply:
Ok will add this in